example/js2/api/ICU4XFixedDecimalFormatter.mjs

```diff
@@ -0,0 +1,81 @@
+// generated by diplomat-tool
+
+import { ICU4XDataProvider } from "./ICU4XDataProvider.mjs"
+import { ICU4XFixedDecimal } from "./ICU4XFixedDecimal.mjs"
+import { ICU4XFixedDecimalFormatterOptions } from "./ICU4XFixedDecimalFormatterOptions.mjs"
+import { ICU4XLocale } from "./ICU4XLocale.mjs"
+import wasm from "./diplomat-wasm.mjs";
+import * as diplomatRuntime from "./diplomat-runtime.mjs";
+
+
+
+/** An ICU4X Fixed Decimal Format object, capable of formatting a [`ICU4XFixedDecimal`] as a string.
+*
+*See the [Rust documentation for `FixedDecimalFormatter`](https://docs.rs/icu/latest/icu/decimal/struct.FixedDecimalFormatter.html) for more information.
+*/
+
+
+const ICU4XFixedDecimalFormatter_box_destroy_registry = new FinalizationRegistry((ptr) => {
+    wasm.ICU4XFixedDecimalFormatter_destroy(ptr);
+});
+
+export class ICU4XFixedDecimalFormatter {
+    // Internal ptr reference:
+    #ptr = null;
+
+    // Lifetimes are only to keep dependencies alive.
+    // Since JS won't garbage collect until there are no incoming edges.
+    #selfEdge = [];
+    
+    
+    constructor(ptr, selfEdge) {
+        
+        this.#ptr = ptr;
+        this.#selfEdge = selfEdge;
+        // Unconditionally register to destroy when this object is ready to garbage collect.
+        ICU4XFixedDecimalFormatter_box_destroy_registry.register(this, this.#ptr);
```

Comment:
Question: Is this part of the code generated conditionally on whether the type is owned? I see previously it did a runtime dynamic check on an `owned` boolean.

Reply:
I believe so, I checked in with @robertbastian to see if this was the kind of behavior we wanted for ownership of opaques. I can give it a more thorough look later, however.

Reply:
Yeah I think in Dart we only register for types that are owned? And all references to them are stashed as edges.